src/client/executor.rs

```diff
@@ -765,6 +799,34 @@ impl Client {
         }
     }
 
+    #[cfg(feature = "csfle")]
+    fn auto_encrypt<'a>(
+        &'a self,
+        csfle: &'a super::csfle::ClientState,
+        command: &'a RawDocument,
+        target_db: &'a str,
+    ) -> BoxFuture<'a, Result<RawDocumentBuf>> {
+        Box::pin(async move {
+            let ctx = csfle
+                .crypt
+                .ctx_builder()
+                .build_encrypt(target_db, command)?;
+            self.run_mongocrypt_ctx(ctx, Some(target_db)).await
```

Comment:
WDYT about passing the `&ClientState` down into this method too? That way we won't need to acquire the lock again, and it also makes it explicit that this function is called when the read lock is held and can therefore not acquire the write lock? Ditto for `auto_decrypt`.